pkg/k8sapi/k8sutils.go

```diff
@@ -66,11 +73,25 @@ func CreateCachedKubeClient(rawK8SClient client.Client, mapper meta.RESTMapper)
 	restCfg.Burst = 100
 
 	vpcCniScheme := runtime.NewScheme()
-	clientgoscheme.AddToScheme(vpcCniScheme)
-	eniconfigscheme.AddToScheme(vpcCniScheme)
+	err = clientgoscheme.AddToScheme(vpcCniScheme)
+	if err != nil {
+		return nil, err
+	}
+	err = eniconfigscheme.AddToScheme(vpcCniScheme)
+	if err != nil {
+		return nil, err
+	}
+
+	// IPAMD only needs to cache pods on this node, so the following selector is used to reduce memory consumption
+	cacheOptions := crcache.Options{Scheme: vpcCniScheme, Mapper: mapper}
+	if nodeName := os.Getenv("MY_NODE_NAME"); nodeName != "" {
+		cacheOptions.SelectorsByObject = map[client.Object]crcache.ObjectSelector{&corev1.Pod{}: {
+			Field: fields.Set{"spec.nodeName": nodeName}.AsSelector(),
```

Comment:
How confident are we on this downward API availability? I am seeing an attempt to reopen an issue on this fieldRef from multiple users
https://github.com/kubernetes/kubernetes/issues/64168